### PR TITLE
Implement the Ruckig plugin

### PR DESCRIPTION
**Pull request by AndyZe**
There are a couple changes here that might be controversial:

- I renamed the `joint_limits` pkg to `limit_enforcement_plugins` to describe it better. Previously, I was confused by `joint_limits` vs `joint_limits_interface`.

- I ended up deleting `on_init()` and `on_enforce()`. `on_init()` saved about 20 lines of code duplication in reading of the joint limits but `on_enforce()` was completely different for the 2 plugins so there seems to be no reason for it. Overall, I think the extra layer of abstraction isn't worthwhile.

I haven't tested this yet.

Please don't squash merge.

### CODE REVIEW

**Comment by destogl** [comment]
TBH, I a not sure how to even merge this. There are so many disruptive changes at the moment. Some of the ideas are very good and I agree with them, but I would like to keep the first version simple. So I'll cherry-pick commits from this branch.

**Comment by AndyZe** [comment]
Um, what? Simplicity was my goal, too. Can you do a code review and tell me what you don't like?

File: joint_limits/src/joint_limiter_interface.cpp [hunk 1]
```diff
@@ -1,72 +0,0 @@
...
-#include <string>
-#include <vector>
-
-#include "joint_limits/joint_limits_rosparam.hpp"
-
-// TODO(anyone): Add handing of SoftLimits
-namespace joint_limits
-{
-template <>
-bool JointLimiterInterface<JointLimits>::init(
```
Comment (destogl) [inline joint_limits/src/joint_limiter_interface.cpp:28]: I am not sure why did you remove this file? It's very useful to have default implementations.

File: limit_enforcement_plugins/src/ruckig_joint_limiter.cpp [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+namespace
+{
+constexpr double DEFAULT_MAX_VELOCITY = 5;       // rad/s
+constexpr double DEFAULT_MAX_ACCELERATION = 10;  // rad/s^2
+constexpr double DEFAULT_MAX_JERK = 20;          // rad/s^3
+}  // namespace
+
+template <>
+bool RuckigJointLimiter<limit_enforcement_plugins::JointLimits>::init(
+  const std::vector<std::string> joint_names, const rclcpp::Node::SharedPtr & node,
```
Comment (destogl) [inline limit_enforcement_plugins/src/ruckig_joint_limiter.cpp:34]: We should keep this implementation outside of here. This will result that each implemented plugin has to take care of reading the parameters. Ideally, users should only implement "enforce()" method. and maybe have short impolementation of "init()" to get some structures set-up.

**Review by destogl — COMMENTED** [review]
So I would proceed as follows:

- keep the name `joint_limits`. I am not very happy about this name and it is possible that the package is merged with `joint_limits_interface`. Finally this package should not have any plugins (or maybe only "SimpleJointLimter") implemented. 
- The Ruckig-plugin is here for now because of testing. We have to move it outside because it will make the whole framework depend on ruckig package.
- I started with 2-layer structure because it was well proven to add intermediate checks and extend the library with user-friendly features. This way hardware interfaces and controllers are implemented.

File: joint_limits/src/joint_limiter_interface.cpp (hunk 1 above)
Comment (AndyZe) [reply joint_limits/src/joint_limiter_interface.cpp:28]: The file is not necessary and I thought `simple_joint_limiter` would be a good example for others to follow.

**Comment by AndyZe** [comment]
OK, so basically you don't like any of the proposed changes and want to keep it 100% the way it was previously. There is not a lot I can do to help in that case.

I would at least suggest renaming `joint_limits` package name to something else. It is way too vague.

**Comment by destogl** [comment]
I merged this manually and find a compromise solution for now. Thanks for support!
